core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/sink/managed/ProgressiveSinkOpDesc.java

```diff
@@ -59,7 +65,7 @@ public class ProgressiveSinkOpDesc extends SinkOpDesc {
     public PhysicalOp getPhysicalOp(WorkflowIdentity workflowId, ExecutionIdentity executionId) {
         // Since during workflow compilation phase, the storage can be null, the writer should also be null
         // the writer will be set properly when workflow execution service receives the physical plan
-        final SinkStorageWriter writer = (storage != null) ? storage.getStorageWriter() : null;
+        final BufferedItemWriter<Tuple> writer = this.storage.write();
```

Comment:
write() function returns a writer is not natural to me.

Reply:
Rename to writer